### PR TITLE
Hi-Fi: Story Archive

### PR DESCRIPTION
**Pull request by isabellberzsenyi**
# This PR contains the hi-fi version of the Story Archive page. 
### To help 2 dependencies have been introduced:
1. `react-moment` aids with date parsing for the story previews
2. `react-responsive` aids by offering media queries that help with choosing to render components depending on the device screen size.

### High level approach:
- Refactor getStoryByPage in `api.js` to allow for the results to be ordered
- Nest `[page].js` in `[order]` directory to take in the ordering parameter
- Move `Story Archive` component into `StoryArchivePage` component
- Helper components created: ArchivePagination, ArchiveSorting and ArchiveStoryPreview

### **When checking out the branch, note that the redirects only work when deployed so go to the url ending `archive/date/1`**
**Please note: on the iPhone SE or smaller the footer changes the sizing, this will be fixed when the footer is created.**

### CODE REVIEW

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
image
On the smallest mobile size (iphoneSE), the header is pushing against the div and there is no right margin anymore.
Try changing the margins using percentages / clamp when it gets to mobile to make sure the all stories header stays more centered
I just played around with your branch and a fix is to wrap the H1Styled in another styled component div. Then to set the margin-left to around -1.5em

File: components/storyArchivePage/archivePagination/ArchivePagination.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          src={icons.FILLED_SHORT_ARROW}
+          disabled={prevDisabled}
+          previous={true}
+        />
+      </PageLink>
+      <P>
+        {page} / {totalPages}
+      </P>
+      <PageLink href={ARCHIVE + (page + 1)}>
+        <ArchiveArrow
```
Comment (josephmannis) [inline components/storyArchivePage/archivePagination/ArchivePagination.js:26]: Can you give this an empty alt text please? `alt=""`

File: components/storyArchivePage/archiveSorting/ArchiveSorting.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  '/archive/' + selectedSortType.toLowerCase() + '/' + 1;
+
+export default function ArchiveSorting({ sortType, order }) {
+  const label = order == DATE ? 'MOST RECENT' : 'A - Z';
+  return (
+    <SortLabelWrapper>
+      <PageLink href={sortUrl(order)}>
+        <SortLabel enabled={sortType == order}>{label}</SortLabel>
+      </PageLink>
+      <SortUnderline
```
Comment (josephmannis) [inline components/storyArchivePage/archiveSorting/ArchiveSorting.js:22]: Empty alt please

File: components/storyArchivePage/StoryArchivePage.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      </H1StyledContainer>
+      <PageDescription
+        description={archiveDescription}
+        arrowText="EXPLORE THE ARCHIVE"
+        img={archiveHeaderImage}
+      />
+      <ArchiveContainer>
+        <SortingPaginationWrapper>
+          <ArchiveSortingWrapper>
+            <ArchiveSorting order={DATE} sortType={sortType} />
```
Comment (josephmannis) [inline components/storyArchivePage/StoryArchivePage.js]: In these components, can you add these labels?
https://accessibilityresources.org/aria-sort

**Review by josephmannis — CHANGES_REQUESTED** [review]
nice! few little thangs

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
For the mobile could you please make the archive photo section closer to the description since its a lot closer on the hi-fi? I think tablet might also need to be a smidge closer. Maybe play around w/ clamp / percentages to make sure its a good proportion on all mobile / tablet sizes 
image 
image
image
image

Everything else looks great though! I'm ready to approve after this change:)

**Review by jzhzhang66 — APPROVED** [review]
Looks good!!!

File: components/storyArchivePage/archiveStoryPreview/ArchiveStoryPreview.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  const formattedDate = moment(storyDate).format('MM/DD');
+  const isTabletOrMobile = useMediaQuery({
+    query: `(max-width: ${breakpointsObj.tabletLg}px)`,
+  });
+
+  return (
+    <StoryPreviewContainer>
+      <StoryPreviewImage src={previewImage.url} alt={previewImage.alt} />
+      <PageLink href={storyUrl}>
+        <>
```
Comment (jean-zhang) [inline components/storyArchivePage/archiveStoryPreview/ArchiveStoryPreview.js]: Do we need this fragment?

File: components/storyArchivePage/archiveStoryPreview/ArchiveStoryPreview.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  const storyUrl = '/story/' + uid;
+  const formattedDate = moment(storyDate).format('MM/DD');
+  const isTabletOrMobile = useMediaQuery({
+    query: `(max-width: ${breakpointsObj.tabletLg}px)`,
+  });
+
+  return (
+    <StoryPreviewContainer>
+      <StoryPreviewImage src={previewImage.url} alt={previewImage.alt} />
+      <PageLink href={storyUrl}>
```
Comment (jean-zhang) [inline components/storyArchivePage/archiveStoryPreview/ArchiveStoryPreview.js:36]: This page link isn't working for me on the deployed version

**Review by jean-zhang — APPROVED** [review]

File: components/storyArchivePage/archiveStoryPreview/ArchiveStoryPreview.js (hunk 4 above)
Comment (isabellberzsenyi) [reply components/storyArchivePage/archiveStoryPreview/ArchiveStoryPreview.js:37]: PageLink only takes in 1 component, so here it's necessary

**Pull request merged**